### DISCUSSION TITLE
Gatsby Plugin Error - Cannot read properties of undefined (reading 'setParserPlugins')

### DISCUSSION THREAD

**Discussion by elbuendiscipulo** [mdx-js/mdx#1847] [Q&A] [answered]
I get this error when building a Gatsby starter kit  on the server for production, I have tested the code on 3 different computers and it works locally on MacOS and Windows, however when on the server (Ubuntu) I cannot build the code. 

This is the error I get, Im a bit new to Gatsby and this plugin, any help is very appreciated. 

<img width="732" alt="Screen Shot 2021-11-30 at 17 58 39" src="https://user-images.githubusercontent.com/94419591/144147147-91c80252-adbf-4bc1-857c-73a59e1e747a.png">

**Comment by ChristianMurphy** [comment, marked as answer]
`gatsby-plugin-mdx` is maintained by Gatsby, looking at their issue tracker https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+setParserPlugins+is%3Aclosed it looks like there were some recently fixed issues for plugin loading.

It may be worth checking that the versions of packages used locally and on the server are the same and are using a recent version of Gatsby and it's plugins.

It would also be worth asking the Gatsby team for their thoughts, as they are the maintainers of the code that appears to be throwing the error https://github.com/gatsbyjs/gatsby/discussions

**Reply by elbuendiscipulo** [reply]
Thank you @ChristianMurphy

**Reply by cameronbraid** [reply]
Any luck with this ?  I just hit the same error

**Reply by cameronbraid** [reply]
Found that some of my versions were too old

https://github.com/gatsbyjs/gatsby/issues/33677#issuecomment-964000968
